include/openssl/ocsp.h

```diff
@@ -164,9 +164,26 @@ OPENSSL_EXPORT int OCSP_resp_find_status(OCSP_BASICRESP *bs, OCSP_CERTID *id,
                                          ASN1_GENERALIZEDTIME **thisupd,
                                          ASN1_GENERALIZEDTIME **nextupd);
 
-// OCSP_basic_verify verifies a basic response message. Returns 1 if the
-// response is valid, 0 if the signature cannot be verified, or -1 on fatal
-// errors such as malloc failure.
+// OCSP_check_validity checks the validity of |thisUpdate| and |nextUpdate|
+// fields from an |OCSP_SINGLERESP|.
+//
+// Note: It is possible that the request will take a few  seconds to process
+// and/or the time won't be totally accurate. Therefore, to avoid rejecting
+// otherwise valid time we allow the times to be within |drift_num_seconds| of
+// the current  time. Also, to avoid accepting very old responses without a
+// |nextUpdate| field, an optional |max_age_seconds| parameter specifies the
+// maximum age the |thisUpdate| field can be. |max_age_seconds| should be the
+// number of seconds relative to |thisUpdate|. You can also set
+// |max_age_seconds| to "-1", if the maximum age should not be checked.
+OPENSSL_EXPORT int OCSP_check_validity(ASN1_GENERALIZEDTIME *thisUpdate,
+                                       ASN1_GENERALIZEDTIME *nextUpdate,
+                                       long drift_num_seconds,
+                                       long max_age_seconds);
+
+// OCSP_basic_verify verifies a basic response message. It checks that |bs| is
```

Comment:
Awesome comment. What is the recommend way to fully verify an OCSP response? Is it call OCSP_basic_verify, if success check OCSP_check_validity, is there another step after that?

Reply:
Yes, after `OCSP_basic_verify`, `OCSP_check_validity` should be used to verify the OCSP response's timestamps. Asides from that, the caller should also use `OCSP_resp_find_status` to verify if the OCSP response's Cert status is `V_OCSP_CERTSTATUS_GOOD`.
I'll add both as `Note 2.` below :) .